providers/flagd/pkg/service/service.go

```diff
@@ -432,6 +421,9 @@ func (s *Service) streamClient(ctx context.Context) error {
 	s.logger.V(logger.Info).Info("connected to event stream")
 
```

Comment:
Based on the docs (see [Creating Providers](https://github.com/open-feature/flagd/blob/main/docs/other_resources/creating_providers.md#provider-lifecycle-initialization-and-shutdown)), I think here we need to know if we previously failed to emit a `Ready` and `Changed` events

Reply:
The initial ready state is emitted from flagd. This happens at the first event stream connection establishment. 

State change : NOT_READY -> READY   (this event is emitted by SDK, flagd provider do not emit this)

From there on, we can get Config changes, Provider shutdown or connection errors , which are handled by the `streamClient ` logic. 

I do not see the need to store the previous state and could rely on current stream message to produce the valid state